src/main/java/org/jabref/logic/openoffice/oocsltext/CSLCitationOOAdapter.java

```diff
@@ -83,11 +296,88 @@ private String transformHtml(String html) {
         // Replace span tags with inline styles for italic
         html = html.replaceAll("<span style=\"font-style: ?italic;?\">(.*?)</span>", "<i>$1</i>");
 
+        // Replace span tags with inline styles for underline
+        html = html.replaceAll("<span style=\"text-decoration: ?underline;?\">(.*?)</span>", "<u>$1</u>");
+
         html = html.replaceAll("<span style=\"font-variant: ?small-caps;?\">(.*?)</span>", "<smallcaps>$1</smallcaps>");
 
         // Clean up any remaining span tags
         html = html.replaceAll("</?span[^>]*>", "");
 
         return html;
     }
+
+    public boolean isCitedEntry(BibEntry entry) {
+        String citationKey = entry.getCitationKey().orElse("");
+        return markManager.hasCitationForKey(citationKey);
+    }
+
+    private String generateAlphanumericCitation(List<BibEntry> entries, BibDatabaseContext bibDatabaseContext) {
+        StringBuilder citation = new StringBuilder("[");
+        for (int i = 0; i < entries.size(); i++) {
+            BibEntry entry = entries.get(i);
+            Optional<String> author = entry.getResolvedFieldOrAlias(StandardField.AUTHOR, bibDatabaseContext.getDatabase());
+            Optional<String> year = entry.getResolvedFieldOrAlias(StandardField.YEAR, bibDatabaseContext.getDatabase());
+
+            if (author.isPresent() && year.isPresent()) {
+                AuthorList authorList = AuthorList.parse(author.get());
+                String alphaKey = authorsAlpha(authorList);
+
+                // Extract last two digits of the year
+                String shortYear = year.get().length() >= 2 ?
+                        year.get().substring(year.get().length() - 2) :
+                        year.get();
+
+                citation.append(alphaKey).append(shortYear);
+            } else {
+                citation.append(entry.getCitationKey().orElse(""));
+            }
+
+            if (i < entries.size() - 1) {
+                citation.append("; ");
+            }
+        }
+        citation.append("]");
+        return citation.toString();
+    }
+
+    private boolean isAlphanumericStyle(CitationStyle style) {
+        return "DIN 1505-2 (alphanumeric, Deutsch) - standard superseded by ISO-690".equals(style.getTitle());
+    }
+
+    public static String authorsAlpha(AuthorList authorList) {
```

Comment:
This method should be part of AuthorLIst, then you can add a proper test

Reply:
There is already an authorsAlpha in BracketedPattern
Since I am using a derived version for CSL, I was planning to keep this one here.
I have plans for refactoring all the formatters into one CSLFormatUtils in next PR, will put this there if alright?

Reply:
Okay, then add this to the todo list for the next PR(s).